### ISSUE TITLE
azurerm_batch_account pool_allocation_mode "UserSubscription" fails

### ISSUE THREAD

**Issue by phosphre** [terraform-providers/terraform-provider-azurerm#2793]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.11.10
+ provider.azurerm v1.21.0
+ provider.random v2.0.0

### Affected Resource(s)

* `azurerm_batch_account`

### Terraform Configuration Files

azurerm_storage_account and azurerm_batch_account substanzas hcl omitted for brevity.

```hcl
resource "azurerm_batch_account" "main" {
  name                 = "phtestbatch"
  resource_group_name  = "rg-dev-ph-test"
  location             = "northeurope"
  pool_allocation_mode = "UserSubscription"
  storage_account_id   = "${azurerm_storage_account.main.id}"

<snip>
```

### Debug Output

1 error(s) occurred:

```hcl
module.demo-batch.azurerm_batch_account.main: 1 error(s) occurred:

azurerm_batch_account.main: Error creating Batch account "phtestbatch" (Resource Group "rg-dev-ph-test"): 
batch.AccountClient#Create: Failure sending request: StatusCode=400 -- 
Original Error: Code="InvalidRequestBody" Message="The specified Request Body is not syntactically valid.\nRequestId:27c0b0f8-f459-ae12-8712-d0c5a084d03\nTime:2019-01-29T10:49:13.3297421Z" 
Target="BatchAccount" Details=[{"code":"Reason","message":"keyVaultReference must be set if poolAllocationMode is specified as 'UserSubscription' on a PUT request"}]
```

Judging from the ref below, there is no need to supply detailed debug.

### Panic Output

N/A

### Expected Behavior

Terraform to error because no references to KeyVault resource supplied in the configuration, or, if one were to be, then success output should have resulted.

### Actual Behavior

Error as per the above debug output.

### Steps to Reproduce

1. Write a `azurerm_batch_account` resource, and specify `pool_allocation_mode = "UserSubscription"`, including a supporting azurerm_storage_account resource
1. `terraform apply`

### Important Factoids

The tests conducted on this resource only covered BatchService not UserSubscription and I am in a corporate environment requiring private node IPs.  I thus need to use `UserSubscription`.

### References

* #2428

**Comment by phosphre** [comment]
Hello any chance this could be fixed for 1.24.0 please?

**Comment by phosphre** [comment]
Hello any chance this could be fixed for 1.28.0 please?

**Comment by jcorioland** [comment]
@tombuildsstuff I will also have a look on this one.

**Comment by phosphre** [comment]
Could I bump this please if it's not too much weight on the 2.0.0 proceedings?

**Comment by mikeclearbank** [comment]
any news on this one?

thanks

**Comment by tombuildsstuff** [comment]
Fixed via #3535 - thanks @jcorioland :)
